plugins/topology/src/__fixtures__/1-deployments.ts

```diff
@@ -2323,6 +2323,2946 @@ export const mockKubernetesResponse = {
+            name: 'APP_NAME',
+            type: 'string',
+          },
+          {
+            default: 'https://github.com/sclorg/nodejs-ex.git',
+            name: 'GIT_REPO',
+            type: 'string',
+          },
+          {
+            default: '',
+            name: 'GIT_REVISION',
+            type: 'string',
+          },
+          {
+            default:
+              'image-registry.openshift-image-registry.svc:5000/deb-test/nodejs-ex-git',
+            name: 'IMAGE_NAME',
+            type: 'string',
+          },
+          {
+            default: '.',
+            name: 'PATH_CONTEXT',
+            type: 'string',
+          },
+          {
+            default: '16-ubi8',
+            name: 'VERSION',
+            type: 'string',
+          },
+        ],
+        tasks: [
+          {
+            name: 'fetch-repository',
+            params: [
+              {
+                name: 'url',
+                value: '$(params.GIT_REPO)',
+              },
+              {
+                name: 'revision',
+                value: '$(params.GIT_REVISION)',
+              },
+              {
+                name: 'subdirectory',
+                value: '',
+              },
+              {
+                name: 'deleteExisting',
+                value: 'true',
+              },
+            ],
+            taskRef: {
+              kind: 'ClusterTask',
+              name: 'git-clone',
+            },
+            workspaces: [
+              {
+                name: 'output',
+                workspace: 'workspace',
+              },
+            ],
+          },
+          {
+            name: 'build',
+            params: [
+              {
+                name: 'IMAGE',
+                value: '$(params.IMAGE_NAME)',
+              },
+              {
+                name: 'TLSVERIFY',
+                value: 'false',
+              },
+              {
+                name: 'PATH_CONTEXT',
+                value: '$(params.PATH_CONTEXT)',
+              },
+              {
+                name: 'VERSION',
+                value: '$(params.VERSION)',
+              },
+            ],
+            runAfter: ['fetch-repository'],
+            taskRef: {
+              kind: 'ClusterTask',
+              name: 's2i-nodejs',
+            },
+            workspaces: [
+              {
+                name: 'source',
+                workspace: 'workspace',
+              },
+            ],
+          },
+          {
+            name: 'deploy',
+            params: [
+              {
+                name: 'SCRIPT',
+                value: 'oc rollout status deploy/$(params.APP_NAME)',
+              },
+            ],
+            runAfter: ['build'],
+            taskRef: {
+              kind: 'ClusterTask',
+              name: 'openshift-client',
+            },
+          },
+        ],
+        workspaces: [
+          {
+            name: 'workspace',
+          },
+        ],
+      },
+    },
+  ],
```

Comment:
I have added 2 PipelineRuns, with different scenarios involving the finally tasks. Also, I had to test if the latest pipeline run is shown in the decorator. So I have updated the tekton resources file that you created. Let me know what you think ?

Reply:
Thanks @debsmita1. Just wanted to highlight this so we keep only the data that is required and use the existing components/utils instead of creating new ones.